### PR TITLE
improvement: Add builder to heartbeat sources

### PR DESCRIPTION
**Pull request by ashrayjain**
## Before this PR
We have constructors for heartbeat sources that need compile breaks / new constructors every time another option needs to added.

## After this PR
==COMMIT_MSG==
Heartbeat sources use the builder pattern which allows easy extensibility in the future.
==COMMIT_MSG==

## Possible downsides?

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/witchcraft-go-health/43)

### CODE REVIEW

File: sources/heartbeat/options.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+func (h *heartbeatSourceConfig) apply(options ...Option) {
+	for _, option := range options {
+		option(h)
+	}
+}
+
+// WithStartupGracePeriod configures an initial grace period that allows the health check source to remain healthy
+// despite no heartbeats within the configured heartbeat timeout. This grace period is only applied once on startup.
```
Comment (tabboud) [inline sources/heartbeat/options.go:45]: Mind adding in that this will default to 0 or no startup grace period if not configured

**Review by tabboud — COMMENTED** [review]
LGTM, just 1 nit on the godoc.

File: sources/heartbeat/options.go (hunk 1 above)
Comment (ashrayjain) [reply sources/heartbeat/options.go:45]: Done

**Review by tabboud — APPROVED** [review]
Approving, but looks like the changelog-bot is having some trouble. It might be best to add one manually, or set `no-changelog`

**Pull request merged**